### PR TITLE
Chrome: Adding the post title input

### PR DESCRIPTION
**Pull request by youknowriad**
Now that we're able to save posts, we need a Post Title input. 
This PR adds this and shows the input in the Visual and the Text Editor.
This PR doesn't include the permalink editor yet.
It probably needs some design polish.

### CODE REVIEW

**Comment by jasmussen** [comment]
Nice!

This really needs placeholder text in order to work. Without it it's kind of hard to see where the input field is. Which I mean, we might want to fix this with an underline border regardless, but it would be good to test with placeholder text first. 

It looks like there was a small regression with the responsive breakpoint around 600px in the text editor:

screen shot 2017-05-08 at 12 06 37

Would you like me to take a look?

**Comment by youknowriad** [comment]
@jasmussen I've tried to fix those. Let me know if it's better now and feel free to make any change here ;)

**Comment by jasmussen** [comment]
Polished the placeholder. Looks good to me now! 👍

File: editor/post-title/index.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+		title: state.editor.edits.title === undefined
+			? state.currentPost.title.raw
+			: state.editor.edits.title,
+	} ),
+	( dispatch ) => {
+		return {
+			onUpdate( title ) {
+				dispatch( {
+					type: 'EDIT_POST',
+					post: { title }
```
Comment (aduth) [inline editor/post-title/index.js]: Not specific to this pull request, but I'm wondering if the `post` key naming here is misleading in how we're only applying changes as a patch, not as a full post object. 

@nylen @youknowriad do you have any thoughts on this? Would another name like `edits`, `values`, `properties` be any clearer?

**Review by aduth — APPROVED** [review]
Looks good 👍

File: editor/post-title/index.js (hunk 1 above)
Comment (youknowriad) [inline editor/post-title/index.js]: Yes, it seemed a bit weird to me too. I would prefer `edits`

**Comment by nylen** [comment]
https://github.com/WordPress/gutenberg/pull/709#discussion_r115279305

> I'm wondering if the `post` key naming here is misleading in how we're only applying changes as a patch, not as a full post object.

> Yes, it seemed a bit weird to me too. I would prefer `edits`

:+1: No objection here

File: editor/post-title/index.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+function PostTitle( { title, onUpdate } ) {
+	const onChange = ( event ) => onUpdate( event.target.value );
+
+	return (
+		<h1 className="editor-post-title">
```
Comment (aduth) [inline editor/post-title/index.js]: Hmm, should this be an `<h1 />` ?

Comment (aduth) [inline editor/post-title/index.js]: I guess it makes sense in the context of being the `<h1 />` of the post itself (title). I think my concern is in the context of the admin screen. Not really sure what the correct answer is here.

Comment (youknowriad) [inline editor/post-title/index.js]: Yeah, the reasoning was that the post title should be `h1` but I agree this is essential for the frontend but probably not for the editor itself.

**Pull request merged**